consensus/test/autonity_interactor_test.go

```diff
@@ -0,0 +1,236 @@
+package test
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"math/big"
+	"strconv"
+
+	"github.com/clearmatics/autonity/accounts/abi/bind"
+	"github.com/clearmatics/autonity/common"
+	"github.com/clearmatics/autonity/contracts/autonity"
+	"github.com/clearmatics/autonity/crypto"
+	"github.com/clearmatics/autonity/ethclient"
+)
+
+// interact is the starting point to constructing a tool to interact with
+// autonity. It accepts an rpc port to connect to a local autonity instance and
+// then, through a sequence of method calls allows interaction with autonity.
+// Any error encountered along the way is propagated to the end of the chain of
+// method calls. All of the components of the system are designed to be single
+// use only.
+//
+// Example usages
+// err := autonityInstance(rpcPort).tx(senderKey).addValidator(address ,stake, enode)
+//
+// whitelist,err := autonityInstance(rpcPort).call(blockNumber).getWhitelist()
```

Comment:
are you sure its not supposed to be:
```suggestion
// err := interact(rpcPort).tx(senderKey).addValidator(address ,stake, enode)
//
// whitelist,err := interact(rpcPort).call(blockNumber).getWhitelist()
```